### DISCUSSION TITLE
Wait for LLDP packet, no way to force TX of LLDP

### DISCUSSION THREAD

**Discussion by toomuchio** [ipxe/ipxe#1347] [General]
I'm wondering is there a way to safely wait until ipxe receives an LLDP packet? Sometimes devices wont transitmit LLDP frames for a while after link has come up which causes isssues if you want to use that value.

There doesn't seem to be a way to TX a dummy LLDP value to force an RX of one either.

I'm trying this at the moment but it doesn't seem to work, get stuck in an endless loop.

```
:lldp_wait
echo Where lldp
sleep 1
isset ${net0.lldp/5.0.0.0} || goto lldp_wait
echo Switch name: ${net0.lldp/5.0.0.0:string}
echo Switch port: ${net0.lldp/5.2.0.1.0:string}
```

**Comment by mcb30** [comment]
As per the LLDP specification: "LLDP does not itself contain a mechanism for soliciting specific information from other LLDP agents".  The protocol does not have a discovery mechanism, so it's up to the switch to choose when to send an LLDP packet and there's no (specified) way for iPXE to influence this.

If your switch is not sending LLDP packets when you expect it to do so, then you probably need to investigate the switch configuration.
